src/main/java/org/apache/sling/distribution/agent/impl/SimpleDistributionAgentQueueProcessor.java

```diff
@@ -147,6 +147,9 @@ private boolean processQueueItem(String queueName, DistributionQueueEntry queueE
                 } catch (RecoverableDistributionException e) {
                     distributionLog.warn("[{}] PACKAGE-FAIL {}: could not deliver {}, {}", queueName, requestId, distributionPackage.getId(), e.getMessage());
                     distributionLog.debug("could not deliver package {}", distributionPackage.getId(), e);
+                    // since there is a recoverable error, it is possible that the same error is observed on the retry
+                    // we should add a linear backoff using random delay before re-attempting to distribute the same item.
+                    addRandomDelay(queueItemStatus.getAttempts());
```

Comment:
Thinking more on this, we'd want to add this to the top of the `try` block - something like:
```
try {
    int processingAttempts = queueItemStatus.getAttempts();
    if (processingAttempts > 0) {
        // this is not the first attempt - throttle
        addRandomDelay(queueItemStatus.getAttempts());
    }
    ... // business as usual
} catch (...) {
...
}
```

I say so because even for non-recoverable exceptions (codified as `catch (Throwable e) {...}`) we should be adding the delay.

Reply:
For non-recoverable exceptions there will be no retries AFAIK. Such items will directly be moved to the error queue.

Reply:
hm, the last two scenarios we faced (which made us re-prio this issue) weren't moved to error-queue (and would only do so now because we patched them upstream).
Essentially, there are cases where we can't rely on the `catch (Throwable e) {}` clause in this `try {}`, or the `if (... != null) { }` prior to it (e.g., an unchecked exception thrown by `DistributionUtils.getResourceResolver` or `distributionPackageExporter.getPackage`). Which is where adding the delay suggested [here](https://github.com/apache/sling-org-apache-sling-distribution-core/pull/49/files#r593188382) would help.
Also, from a semantic PoV, doing so would mean "we don't care _what_ caused the queue-item to have been retried, but since we _are_ seeing a retry, let's throttle it" - which aiu is what we want.

Do you see any disadvantage if we were to move the `addRandomDelay` call to the top of the `try` block?

Reply:
@actinium15 I thought you were suggesting to move this at the top of the inner try block (because of the catch block you added in the example code). The inner try block only tries to distribute a valid distribution package. However, we can add this delay (based on greater than zero retries) at the top of the outer try block. I will update the PR accordingly.